clients/unshaded/src/main/java/tachyon/client/BlockOutStream.java

```diff
@@ -58,12 +58,14 @@ public static BlockOutStream get(TachyonFile tachyonFile, WriteType opType, int
    */
   public static BlockOutStream get(TachyonFile tachyonFile, WriteType opType, int blockIndex,
       long initialBytes, TachyonConf tachyonConf) throws IOException {
-
-    if (tachyonFile.mTachyonFS.hasLocalWorker()
-        && tachyonConf.getBoolean(Constants.USER_ENABLE_LOCAL_WRITE,
-            Constants.DEFAULT_USER_ENABLE_LOCAL_WRITE)) {
+    if ((tachyonFile.mTachyonFS.hasLocalWorker()
+            && tachyonConf.getBoolean(Constants.USER_ENABLE_LOCAL_WRITE,
+            Constants.DEFAULT_USER_ENABLE_LOCAL_WRITE)
+            && tachyonConf.get(
```

Comment:
Ahhh, I see. Then, the issue is that hasLocalWorker() is wrong now, since the LocalFirst strategy now can return a remote worker. I think the correct way to do this is to fix hasLocalWorker() to correctly return true when it is connected to a local worker. Then, this part of the logic can remain unchanged.